libr/bin/bin.c

```diff
@@ -1099,17 +1100,18 @@ R_API void r_bin_list_archs(RBin *bin, int mode) {
 				pj_ki (pj, "bits", bits);
 				pj_kn (pj, "offset", boffset);
 				pj_kn (pj, "size", obj_size);
-				if (!strcmp (arch, "mips") && h_flag) {
-					pj_ks (pj, "isa", h_flag);
+				if (!strcmp (arch, "mips")) {
+					pj_ks (pj, "isa", info->cpu);
+					pj_ks (pj, "features", info->features);
 				}
 				if (machine) {
 					pj_ks (pj, "machine", machine);
 				}
 				pj_end (pj);
 				break;
 			default:
-				str_fmt = h_flag && strcmp (h_flag, "unknown_flag")? sdb_fmt ("%s_%i %s", arch, bits, h_flag) \
-						  : sdb_fmt ("%s_%i", arch, bits);
+				str_fmt = strcmp (h_flag, "unknown_flag")? sdb_fmt ("%s_%i %s", arch, bits, h_flag) \
```

Comment:
you removed the `h_flag &&` check, which was required to avoid NULL dereferences.